moped-editor/src/views/projects/projectView/ProjectActivityLog.js

```diff
@@ -182,10 +178,24 @@ const ProjectActivityLog = () => {
     data["moped_phases"].forEach((phase) => {
       phaseList[`${phase.phase_id}`] = phase.phase_name;
     });
+    data["moped_tags"].forEach((tag) => {
+      tagList[`${tag.id}`] = tag.name;
+    });
   }
 
+  const selectEntryComponent = (change) => {
+    switch (change.record_type) {
+      case "moped_project":
+        return <ProjectActivityEntry change={change} />;
+      case "moped_proj_tags":
+        return <TagsActivityEntry change={change} tagList={tagList} />;
+      default:
+        return <div>There is no matching component</div>;
+    }
+  };
```

Comment:
Seeing this, I'm 1,000% sold behind on your idea of how to structure this into various components. 🎯 Good call!